packages/patternfly-4/react-core/src/components/Title/Title.js

```diff
@@ -19,18 +19,34 @@ const propTypes = {
   /** content rendered inside the Title */
   children: PropTypes.node,
   /** additional classes added to the Title */
-  className: PropTypes.string
+  className: PropTypes.string,
+  /** the heading level to use */
+  headingLevel: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6'])
 };
 
 const defaultProps = {
   children: '',
-  className: ''
+  className: '',
+  headingLevel: 'h1'
 };
 
-const Title = ({ size, className, children, ...props }) => (
-  <h1 {...props} className={css(styles.title, getModifier(styles, size), className)}>
-    {children}
-  </h1>
+const getTitleElement = (level, children, ...props) => {
+  return React.createElement(level, ...props, children);
+};
+
+const Title = ({ size, className, children, headingLevel, ...props }) => (
```

Comment:
Also, there already is a size prop and the headingLevel does not have a visual impact. So this is just to change it in the DOM? Trying to understand the need, thanks

Reply:
@jschuler there can only ever be 6 possible values here, I think it's best to limit the options to one of those. Allowing an arbitrary component to be used as the base element opens the door to invalid markup and/or incompatible combinations of props.

The size prop is used to determine the visual impact while the headingLevel prop can now be used to control the semantics. Users need to be able to utilize the full range of possible heading levels so they can specify titles that fit into the their app-specific IA.

Does that make sense?

Reply:
Sounds good about the semantics reasoning. Maybe I am missing something here but I don't see any checks for the component. In the example i can add headingLevel="asd" and it will render. I think you'd have the same developer console warnings if you directly used Component

Reply:
Yea I noticed oneOf prop type allows most element-looking string values to be used but throws an error in the console if it's not one of the explicitly defined values. The error that's generated is pretty useful, and is what we would want, no?

<img width="973" alt="screen shot 2018-12-05 at 9 21 07 am" src="https://user-images.githubusercontent.com/5942899/49519746-b960fe80-f86f-11e8-9f6b-c16dac3ec40e.png">

This seems like a textbook use case for oneOf prop type to me. Is there a scenario where this breaks down and using Component becomes necessary?

What kind of checks do you mean? Happy to add whatever is needed, feel free to point me to an example if it's easier!

Reply:
After speaking more offline, I think I was misunderstanding the recommendation here. We can still specify the oneOf prop type and avoid using React.createElement if we reference the value as uppercase in jsx. I was initially trying to do exactly that, just didn't realize my lowercase prop name was the problem. Will update this PR with those changes so the general pattern used is the same as found elsewhere in PF.